src/status_im2/contexts/quo_preview/selectors/react.cljs

```diff
@@ -0,0 +1,79 @@
+(ns status-im2.contexts.quo-preview.selectors.react
+  (:require [clojure.string :as string]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [status-im2.constants :as constants]
+            [status-im2.contexts.quo-preview.preview :as preview]))
+
+(defn- gen-quantity
+  [max-count _]
+  (rand-int max-count))
+
+(def ^:private memo-gen-quantity (memoize gen-quantity))
+
+(def ^:private descriptor
+  [{:label "Add reaction"
+    :key   :add-reaction?
+    :type  :boolean}
+   {:label   "Reactions"
+    :key     :reaction-ids
+    :type    :multi-select
+    :options (for [[reaction-id reaction] constants/reactions]
+               {:key   reaction-id
+                :value (string/capitalize
+                        (last
+                         (string/split (name reaction) #"/")))})}
+   {:label "Max rand. reactions (helper)"
+    :key   :max-count
+    :type  :number}
+   {:label "Pinned"
+    :key   :pinned?
+    :type  :boolean}
+   (preview/customization-color-option
+    {:label "Pinned BG (test)"})])
+
+(defn preview-react
+  []
+  (let [state               (reagent/atom {:add-reaction?       true
+                                           :max-count           1000
+                                           :reaction-ids        [1 2 3]
+                                           :customization-color :blue
+                                           :pinned?             false})
+        reaction-ids        (reagent/cursor state [:reaction-ids])
+        add-reaction?       (reagent/cursor state [:add-reaction?])
+        max-count           (reagent/cursor state [:max-count])
+        pinned?             (reagent/cursor state [:pinned?])
+        customization-color (reagent/cursor state [:customization-color])
+        pressed-reactions   (reagent/atom #{1})
+        reactions           (reagent/reaction
+                             (mapv #(assoc {}
+                                           :emoji-id %
+                                           :quantity (memo-gen-quantity @max-count %)
+                                           :own      (boolean (some (fn [v] (= v %))
```

Comment:
nit: not sure but I think ` :own  (some = @pressed-reactions)` will also work

Reply:
yep, pretty sure it should. Thanks! 👍

Reply:
@J-Son89 I think your suggestion won't work because we expect to pass at least two parameters to `=`, and `some` will only inject one.

 @clauxx 
ok, sorry in advance for being that guy that complains about code :sweat_smile:  but particulary in this part, I think we are doing unnecessary work:

I added some notes to the code:
```clojure
(reagent/reaction ;; 1. do we really need a reaction?
   (mapv #(assoc {} ;; 2. assoc to an empty map, better to just create the map
            :emoji-id %
            :quantity (memo-gen-quantity @max-count %)
            :own      (boolean (some (fn [v] (= v %)) ;; 3. boolean on an already boolean value.
                                     @pressed-reactions)))
         @reaction-ids))
```

so,
1. Why are we using a reaction? could you please explain why is it required or what's the benefit of it?

2. We can use the `hash-map` function to create a map:
    ```clojure
    (mapv #(hash-map ;; instead of assoc to `{}`
            :emoji-id %
            :quantity (memo-gen-quantity @max-count %)
            :own      (boolean (some (fn [v] (= v %))
                                         @pressed-reactions)))
          @reaction-ids)
    ```

    But I'd prefer to use `{}` and give a name to the parameter, since we are using it 3 times in the body, so:
    ```clojure
    (mapv (fn [reaction-id]
            {:emoji-id reaction-id
             :quantity (memo-gen-quantity @max-count reaction-id)
             :own      (boolean (some (fn [v] (= v reaction-id))
                                      @pressed-reactions))})
          @reaction-ids)
    ```
    (and also they have the same amount of lines and it's easier to read)
    
3.  about that `some`:
    ```clojure
     (boolean (some (fn [v] (= v reaction-id))
                   @pressed-reactions))
    ```
    `some` will return the fist logical `true` value found, and we are using `=` inside, unless we are making a difference between `nil` and `false` in the component we are using `own`, we can remove `boolean`, and we can use `#()`:
    ```clojure
    (some #(= % reaction-id) @pressed-reactions)
    ```
    but if we look closer at the value of `@pressed-reactions`, we notice it's declared as an atom with a `set`, and we are looking for a key inside it, so we can use [`contains?`](https://clojuredocs.org/clojure.core/contains_q)  which is more idomatic, returns `true` or `false` and doesn't performs a linear search (as we are doing with `some`):

    ```clojure
    (contains? @pressed-reactions reaction-id)
    ```

Reply:
@ulisesmac this is great! thanks for the detailed explanation! 
1. It's not needed anymore. I used the reaction initially b/c it seemed like the right way to combine multiple atoms. Without it `reactions` would not ... react 🙄 to changes from `pressed-reactions`, but just moving that part inside the (fn [] ...) fixed it.
2. perfect. It's much cleaner now!
3. lmao that was the only reason I made it a `set`, but forgot to use it here and only used it lower in the file 🙄. Thanks!

Also, since I'm still learning clojure, all code-improvement comments are **incredibly** useful for learning, so feel free.